### PR TITLE
Updating docker setup

### PR DESCRIPTION
**Pull request by MuruganChandrasekar**
### The below changes were made:

- Included auxiliary backup container(cron-backup) for providing backup support for Nginx, Node-red and Grafana containers.

- Updated the base images used in all Dockerfile from bionic to focal.
- Added Mosquitto(MQTT client) Ubuntu ppa repository to install the latest version and fixed ownership issue when accessing Let's encrypt certs.
- Added TLS/SSL based SMTP authentication support in Postfix container.
- Some minor changes in the following files: Dockerfile, docker-compose.yml, setup.md and shell scripts.

### CODE REVIEW

File: cron-backup/nodered_backup.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        echo "DATE:" "$DATE" > /node_graf_nginx.txt
+        echo " " >> /node_graf_nginx.txt
+        echo "DESCRIPTION: ${SOURCE_NAME}_Nodered backup" >> /node_graf_nginx.txt
+        echo " " >> /node_graf_nginx.txt
+        echo "STATUS: Nodered backup is Failed." >> /node_graf_nginx.txt
+        echo " " >> /node_graf_nginx.txt
+        echo "Something went wrong, Please check it"  >> /node_graf_nginx.txt
+        < /node_graf_nginx.txt mail -s "${SOURCE_NAME}: Nodered Data Backup" "${CRON_BACKUP_MAIL}"
+fi
+< /node_graf_nginx.txt mail -s "${SOURCE_NAME}: Nodered Data Backup" "${CRON_BACKUP_MAIL}"
```
Comment (terrillmoore) [inline cron-backup/nodered_backup.sh:53]: Not a great idea to do this with a file in the root directory. Should use `/tmp` or similar; root directory should not be used as a temporary directory. Can be fixed in next release.

File: docker-compose.yml [hunk 2]
```diff
@@ -3,22 +3,22 @@
 # Name: docker-compose.yml
 #
 # Function:
-# 	Configure the docker-iot-dashboard collection of docker containers.
+#   Configure the docker-iot-dashboard collection of docker containers.
 #
 # Copyright:
-#	This file copyright (c) 2017-2020 by
+#   This file copyright (c) 2017-2020 by
```
Comment (terrillmoore) [inline docker-compose.yml:9]: Year on copyright should be 2017-2021.

File: docker-compose.yml [hunk 3]
```diff
@@ -3,22 +3,22 @@
 # Name: docker-compose.yml
 #
 # Function:
-# 	Configure the docker-iot-dashboard collection of docker containers.
```
Comment (terrillmoore) [inline docker-compose.yml:6]: I don't know why the indentation changed. If you substituted spaces for tabs, you had tab stop set incorrectly. When you change tabs to spaces, it's important to maintain visual consistency, unless there's a good reason not to.

File: mqtts/Dockerfile [hunk 4]
```diff
@@ -6,24 +6,15 @@
 
 # passing arguments to map letsencrypt certs
```
Comment (terrillmoore) [inline mqtts/Dockerfile:7]: This comment looks wrong because you removed lets encrypt.

**Review by terrillmoore — APPROVED** [review]
OK to merge. Comments can be addressed in future.

File: docker-compose.yml (hunk 3 above)
Comment (MuruganChandrasekar) [reply docker-compose.yml:6]: I shouldn't have changed the format. It's corrected now.

File: docker-compose.yml (hunk 2 above)
Comment (MuruganChandrasekar) [reply docker-compose.yml:9]: it's changed.

File: mqtts/Dockerfile (hunk 4 above)
Comment (MuruganChandrasekar) [reply mqtts/Dockerfile:7]: Yes. I removed it.

File: cron-backup/nodered_backup.sh (hunk 1 above)
Comment (MuruganChandrasekar) [reply cron-backup/nodered_backup.sh:53]: Sure. I'll change it in all backup scripts.

**Pull request merged**